test/contrib/test_socks.py

```diff
@@ -199,6 +206,8 @@ class TestSocks5Proxy(IPV4SocketDummyServerTestCase):
     Test the SOCKS proxy in SOCKS5 mode.
     """
     def test_basic_request(self):
+        disable_warnings(NoIPv6Warning)
```

Comment:
This repeated warning disabling is potentially a problem. Does py.test restore the warnings filters after every test run? If not, we're just repeatedly disabling the `NoIPv6Warning`.